docs/upgrading.md

```diff
@@ -5,6 +5,15 @@ the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summar
 
 ## Upgrading to v3.6
 
+### Deprecations
```

Comment:
This should similarly follow order of importance like with https://github.com/argoproj/argo-workflows/pull/13720#discussion_r1790903329 et al. Deprecations would be less important than breakage